src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,46 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static nature() {
+    this.alive = this.alive.filter((dead) =>
+      dead.health !== 0);
+  }
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+  }
 }
+Animal.alive = [];
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name, health = 100, toHide = false) {
+    super(name, health);
+    this.hidden = toHide;
+
+    Animal.alive.push(this);
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+
+    Animal.alive.push(this);
+  }
+  bite(mammal) {
+    if (mammal.hidden === false) {
+      mammal.health -= 50;
+
+      if (mammal.health === 0) {
+        Animal.nature();
```

Comment:
Instead of calling `Animal.nature()` inside the `bite` method, consider using a more efficient approach to remove the dead animal from the `Animal.alive` array. According to the checklist, you should avoid nesting loops and use `filter` instead. Implement a method to update the `Animal.alive` array without using `indexOf` and `splice`.